frontend/libs/studio-components/src/components/StudioNumberInput/StudioDecimalInput.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Textfield, TextfieldProps } from '@digdir/design-system-react';
+import { useTranslation } from 'react-i18next';
+
+export interface StudioDecimalInputProps extends Omit<TextfieldProps, 'onChange'> {
+  description: string;
+  onChange: (value: number) => void;
+  value?: number;
+}
+
+export const StudioDecimalInput = ({
+  description,
+  onChange,
+  value,
+  ...rest
+}: StudioDecimalInputProps) => {
+  const { t } = useTranslation();
+  const [inputValue, setInputValue] = useState('');
+  const [hasBeenBlurred, setHasBeenBlurred] = useState(false);
+  const isEmpty = inputValue === '';
+
+  useEffect(() => {
+    if (!isEmpty && isStringValidDecimalNumber(inputValue)) {
+      setInputValue(convertNumberToString(value));
+    }
+  }, [value]);
```

Comment:
Her tar jeg selvkritikk siden dette var mitt kodeforslag. Linteren sier her at vi må ha med alle avhengighetene til funksjonen. Funksjonsmessig vil det sannsynligvis ikke utgjøre noen forskjell akkurat her, men det er fint å være på den sikre siden. Forholder vi oss til dette, forsikrer vi oss også mot feil når det kommer endringer i fremtiden.
```suggestion
  }, [value, isEmpty, setInputValue]);
```